### PR TITLE
Download istioctl and export to EnvVar

### PR DESCRIPTION
**Pull request by jeremyharisch**
**Description**
The reconciler now uses `istioctl` to install istio on a cluster. Before it was needed to have `istioctl` installed and then set `ISTIOCTL_PATH` pointing to the istioctl binary.
With this PR, the CLI downloads the needed istioctl version, unwraps the binary, and exports the env var.

Changes proposed in this pull request:

- Download the correct istioctl (corresponding to system OS and architecture)
- Only download if wanted istio version is not present in kymaHome folder
- Extracts binary
- Set `ISTIOCTL_PATH` to point to binary
- Unit tests
- Due to our Sprint planning, unit test for windows (`func unzip`) is not done and this feature was not tested on a windows machine
**Related issue(s)**

[Card Link](https://github.com/orgs/kyma-project/projects/7#card-69381774)

### CODE REVIEW

**Review by lindnerby — DISMISSED** [review]

File: internal/istio/istio.go [hunk 1]
```diff
@@ -0,0 +1,387 @@
...
+		winBinName:     winBinName,
+		downloadUrl:    downloadUrl,
+		tarGzName:      tarGzName,
+		tarName:        tarName,
+		zipName:        zipName,
+	}
+}
+
+func (i *Installation) Install() error {
+	// Set OS Version
```
[the remark this answers is not in the document]
Comment (jeremyharisch) [reply internal/istio/istio.go:93]: Oke, I will remove them. It was just easier while developing. I was shifting a lot around in the beginning XD

File: internal/istio/istio.go [hunk 2]
```diff
@@ -0,0 +1,387 @@
...
+		return false, nil
+	}
+	return false, err
+}
+
+func (i *Installation) setOS() {
+	// Get OS Version
+	i.osExt = runtime.GOOS
+	switch i.osExt {
+	case "windows":
```
[the remark this answers is not in the document]
Comment (jeremyharisch) [reply internal/istio/istio.go:162]: Could make sense, I will change it 👍🏻 (But its darwin and not osx)

File: internal/istio/mock/Chart.yaml [hunk 3]
```diff
@@ -0,0 +1,18 @@
+apiVersion: v1
```
[the remark this answers is not in the document]
Comment (jeremyharisch) [reply internal/istio/mock/Chart.yaml]: Very good hint! Appreciate 👍🏻

File: internal/istio/istio.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (lindnerby) [reply internal/istio/istio.go]: Ye, also my example could be better in terms of wrapping consts together :D 
```go
const (
  Windows OS = "windows"
  OSX OS = "darwin"
  Linux OS = "linux"
)
```

**Review by lindnerby — DISMISSED** [review]

File: internal/istio/istio.go [hunk 4]
```diff
@@ -0,0 +1,415 @@
...
+	if deleteSource {
+		if err := os.Remove(source); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func unZip(source, target string, deleteSource bool) error {
+	// TODO Windows: Test + Unit Tests
```
Comment (lindnerby) [inline internal/istio/istio.go:352]: Is this still to do?

**Review by lindnerby — APPROVED** [review]

**Review by skhalash — APPROVED** [review]

**Pull request merged**